scalafix-tests/src/test/resources/checkSyntax/RemoveXmlLiterals.source

```diff
@@ -0,0 +1,112 @@
+rewrites = [RemoveXmlLiterals]
+
+<<< don't add import when not needed
+object A {}
+>>>
+object A {}
+
+<<< single line, no splice
+object A {
+  <div></div>
+}
+>>>
+import scala.xml.quote._
+object A {
+  xml"<div></div>"
+}
+
+<<< single line, triple quoted
+object A {
+  val a = <div b="Hello"/>
+  val b = <a>\</a>
+}
+>>>
+import scala.xml.quote._
+object A {
+  val a = xml"""<div b="Hello"/>"""
+  val b = xml"""<a>\</a>"""
+}
+
+<<< single line, splice
+object A {
+  val bar = "bar"
+  <div>{bar}</div>
+}
+>>>
+import scala.xml.quote._
+object A {
+  val bar = "bar"
+  xml"<div>${bar}</div>"
+}
+
+<<< multi-line, no splice
+object A {
+  val foo =
+    <div>
+      <span>Hello</span>
+    </div>
+}
+>>>
+import scala.xml.quote._
+object A {
+  val foo =
+    xml"""<div>
+      <span>Hello</span>
+    </div>"""
+}
+
+<<< multi-line, splice
+object A {
+  val foo =
+    <div>
+      <span>{"Hello"}</span>
+    </div>
+}
+>>>
+import scala.xml.quote._
+object A {
+  val foo =
+    xml"""<div>
+      <span>${"Hello"}</span>
+    </div>"""
+}
+
+<<< SKIP splice in attribute position
+object A {
+  <foo bar={"Hello"}/>
+}
+>>>
+import scala.xml.quote._
+object A {
+  xml"<foo bar=${"Hello"}/>"
+}
+
+<<< nested xml literals
+object A {
+  <a>{<a>{"Hello"}</a>}</a>
+}
+>>>
+import scala.xml.quote._
+object A {
+  xml"<a>${xml"<a>${"Hello"}</a>"}</a>"
+}
+
+<<< protect $
+object A {
+  <div>$</div>
+}
+>>>
+import scala.xml.quote._
+object A {
+  xml"<div>$$</div>"
+}
+
+<<< SKIP protect curly brace
+object A {
+  <div>{{</div>
+}
+>>>
+import scala.xml.quote._
+object A {
+  xml"<div>{</div>"
```

Comment:
This test is skipped. As I said previously:

The issue is that there are many places in xml literals where braces don't need to be escaped:
```scala
scala> <a b="{{"/>
res0: scala.xml.Elem = <a b="{{"/>

scala> <![CDATA[{{]]> 
res1: scala.xml.PCData = <![CDATA[{{]]>

scala> <xml:unparsed>{{</xml:unparsed>
res2: scala.xml.Unparsed = {{

scala> <?foo {{?>
res3: scala.xml.ProcInstr = <?foo {{?>
```
We would need to parse the xml literal to implement this rewrite